CMakeLists.txt

```diff
@@ -22,6 +22,16 @@ cmake_minimum_required(VERSION 3.10.2)
 
 # NONE = this project has no language toolchain requirement.
 project(Vulkan-Headers NONE)
+file(READ "include/vulkan/vulkan_core.h" ver)
+string(REGEX MATCH "#define VK_HEADER_VERSION_COMPLETE VK_MAKE_VERSION\\(([0-9]+), ([0-9]+), VK_HEADER_VERSION\\)" _ ${ver})
```

Comment:
I'm a little worried that these regexes are whitespace-sensitive.  I believe it's legal to have whitespace between `#` and `define`, and an error as minor as an extra space after `VK_HEADER_VERSION_COMPLETE` would not affect compilation but would break this process.

Reply:
improved the regex to be more robust. I left out the check for space between `#` and `define`, and I hope that no-one will use this capability.

I also added checks to see if one of the regex did not match and write a `FATAL_ERROR`